lib/mu_search/elastic.rb

```diff
@@ -3,25 +3,34 @@
 require 'connection_pool'
 
 # monkeypatch "authentic product check"" in client
-module Elasticsearch
-  class Client
-    alias original_verify_with_version_or_header verify_with_version_or_header
+module ElasticsearchMonkeyPatch
+  private
 
-    def verify_with_version_or_header(...)
-      original_verify_with_version_or_header(...)
-    rescue Elasticsearch::UnsupportedProductError
-      # silenty ignore this error
+  def verify_elasticsearch(*args, &block)
+    while not @verified do
+      sleep 1
+      begin
+        response = @transport.perform_request(*args, &block)
+        response.headers['x-elastic-product'] = 'Elasticsearch'
+        @verified = true
+      rescue StandardError => e
+        Mu::log.info("SETUP") { "no reaction from elastic, retrying..." }
+        next
+      end
     end
+    response
   end
 end
 
+Elasticsearch::Client.prepend(ElasticsearchMonkeyPatch)
+
 # A wrapper around elasticsearch client for backwards compatiblity
 # see https://rubydoc.info/gems/elasticsearch-api/Elasticsearch
 # and https://www.elastic.co/guide/en/elasticsearch/client/ruby-api/current/examples.html
 # for docs on the client api
 ##
 module MuSearch
-  class Elastic
+  class ElasticWrapper
```

Comment:
to avoid conflict with the Elastic::Transport that replaces Elasticsearch::Transport